eta/core/primitives.py

```diff
@@ -66,35 +66,51 @@ def process(
                     continue
 
                 # Convert to polar coordinates
-                mag, ang = cv2.cartToPolar(
-                    flow_cart[..., 0], flow_cart[..., 1])
-                flow_polar = np.dstack((mag, ang))
+                flow_polar = self.cart_to_polar(flow_polar)
 
                 if polar_path:
                     # Write polar fields
                     np.save(polar_path % p.frame_number, flow_polar)
 
                 if vid_path:
                     # Write flow visualization frame
-                    p.write(_polar_flow_to_img(mag, ang))
+                    p.write(_polar_flow_to_img(flow_polar))
 
-    def _process_frame(self, img):
-        '''Processes the next frame.
+    def process_frame(self, img):
+        '''Computes the dense optical flow field for the next frame.
 
         Args:
-            img: the next frame
+            img: an m x n x 3 image
 
         Returns:
-            flow: the optical flow for the frame in Cartesian coordinates
+            flow: an m x n x 2 array containing the optical flow vectors
+                in Cartesian (x, y) format
         '''
-        raise NotImplementedError("subclass must implement _process_frame()")
+        raise NotImplementedError("subclass must implement process_frame()")
 
-    def _reset(self):
+    def reset(self):
         '''Prepares the object to start processing a new video.'''
         pass
 
+    @staticmethod
+    def cart_to_polar(flow_cart):
+        '''Converts the Cartesian optical flow vectors to polar form.
```

Comment:
Brian, I wonder why you make cart_to_polar a static method of the class this time? Any purpose?

Reply:
I made it a static method of the optical flow class so its clear that this functionality is related specifically to optical flow. However, upon second thought, I should probably just make it a top-level method so the user doesn't have to write `primitives.DenseOpticalFlow.cart_to_polar(...)`, which is a mouthfull.